eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1944,6 +1936,14 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
     stay
   }
 
+  def watchFundingTx(commitments: Commitments, additionalKnownSpendingTxs: Seq[ByteVector32] = Nil): Unit = {
+    // TODO: should we wait for an acknowledgment from the watcher?
+    val knownSpendingTxs: Seq[ByteVector32] = List(commitments.localCommit.publishableTxs.commitTx.tx.txid, commitments.remoteCommit.txid) ++ commitments.remoteNextCommitInfo.left.toSeq.map(_.nextRemoteCommit.txid) ++ additionalKnownSpendingTxs
```

Comment:
nit: probably worth using a `Set` here, otherwise the commit txs will be duplicated (they are in `DATA_CLOSING.spendingTxes` and added manually here.

Reply:
Indeed duplicates don't make sense for hints. Done with fa8ceac59.